src/design-system/components/icon-button/IconButton.tsx

```diff
@@ -0,0 +1,68 @@
+import {
+  CircularProgress,
+  IconButtonProps,
+  IconButton as MuiIconButton
+} from '@mui/material'
+import React, { useState } from 'react'
+import { IconButtonVariant } from './IconButton.constants'
+import AddRoundedIcon from '@mui/icons-material/AddRounded'
+import { cn } from '~/utils/cn'
+import './IconButton.scss'
+
+interface S2SIconButtonProps extends Omit<IconButtonProps, 'size'> {
+  variant?: IconButtonVariant
+  size?: 'xs' | 'sm' | 'md' | 'lg'
+  loading?: boolean
+  disabled?: boolean
+  toggleAble?: boolean
+}
+
+export const IconButton: React.FC<S2SIconButtonProps> = ({
+  variant = IconButtonVariant.Primary,
+  size = 'md',
+  loading = false,
+  disabled = false,
+  toggleAble = false,
+  ...props
+}) => {
+  const [isToggled, setIsToggled] = useState<boolean>(false)
```

Comment:
All basic components must be stateless and all logic must be implemented at the level of the parent component (the value and the function to change this value must be passed through props)

Reply:
Then you can edit the description of the issue, here the component is quite limited because handleClick only changes the behavior of the component itself, but this function can perform more difficult logic (for example, API request, interaction with other components or states, etc.). That is why the component should not have its own state and any other logic,
because now the only thing it can do is simply change the state of isTogled and make the appropriate style changes